### PR TITLE
refactor!: align namespaces to singular

### PR DESCRIPTION
**Pull request by kyle-cochran**
Align namespacing in Astro to use singular naming. Also expand the top-level namespace in Cpp

Changes:
`astro` -> `astrodynamics` (Cpp change only)
`trajectory::orbit::models` -> `trajectory::orbit::model`
`trajectory::models` -> `trajectory::model`
`flight::profile::models` -> `flight::profile::model`
`trajectory::state::coordinatessubsets` -> `trajectory::state::coordinatesubset`
`trajectory::state::CoordinatesSubset` -> `trajectory::state::CoordinateSubset`
`conjunction::messages` -> `conjunction::message`

### CODE REVIEW

File: bindings/python/docs/Validation-Analysis/Orbit Propagation Validation Testing RMS Error.ipynb
Comment (kyle-cochran) [inline bindings/python/docs/Validation-Analysis/Orbit Propagation Validation Testing RMS Error.ipynb:1]: Gotta figure out how to get rid of the changes on this one

Comment (vishwa2710) [reply bindings/python/docs/Validation-Analysis/Orbit Propagation Validation Testing RMS Error.ipynb:1]: Honestly these are not really up to date. We can do that as a separate MR.

Comment (apaletta3) [reply bindings/python/docs/Validation-Analysis/Orbit Propagation Validation Testing RMS Error.ipynb:1]: yeah they are already broken af

File: README.md [hunk 1]
```diff
@@ -111,24 +111,24 @@ The library exhibits the following detailed and descriptive structure:
 ├── Trajectory
```
Comment (apaletta3) [inline README.md:111]: Can you update this tree structure to reflect the current repo structure?

**Review by vishwa2710 — APPROVED** [review]

**Pull request merged**